common/prometheus-server/templates/alerts/_prometheus.alerts.tpl

```diff
@@ -171,69 +310,140 @@ groups:
-      severity: info
-      meta: 'Prometheus `{{`{{ $labels.prometheus }}`}}` queueing notifications'
+      severity: warning
+      meta: Prometheus `{{`{{ $labels.prometheus }}`}}` sends a high number of alerts.
     annotations:
-      description: 'Prometheus `{{`{{ $labels.prometheus }}`}}` is backlogging on the notifications queue. The queue has not been empty for 10 minutes. Current queue length `{{`{{ $value }}`}}`.'
-      summary: Prometheus is queueing notifications.
+      description: |
+        Prometheus `{{`{{ $labels.prometheus }}`}}` sends
+        a high number of alerts to alert managers.
+        <https://{{ include "prometheus.externalURL" . }}/graph?g0.expr={{ urlquery `topk(5, count by (alertname) (ALERTS))` }}|Affected alerts >
+      summary: Prometheus has encountered a high number of alerts sent to alert managers.
+
+  - alert: PrometheusNotificationQueueRunningFull
+    # Without min_over_time, failed scrapes could create false negatives, see
+    # https://www.robustperception.io/alerting-on-gauges-in-prometheus-2-0 for details.
+    expr: |
+      (
+        predict_linear(prometheus_notifications_queue_length{prometheus="{{ include "prometheus.name" . }}"}[5m], 60 * 30)
+      >
+        min_over_time(prometheus_notifications_queue_capacity{prometheus="{{ include "prometheus.name" . }}"}[5m])
+      )
+    for: 15m
+    labels:
+      service: {{ default "metrics" $root.Values.alerts.service }}
+      support_group: {{ default "observability" $root.Values.alerts.support_group }}
+      severity: warning
+      meta: Prometheus `{{`{{ $labels.prometheus }}`}}` alert notification queue predicted to run full in less than 30m.
+    annotations:
+      description: |
+        Alert notification queue of Prometheus `{{`{{ $labels.prometheus }}`}}`
+        is running full.
+      summary: Prometheus alert notification queue predicted to run full in less than 30m.
   {{- end }}
 
   {{- if $root.Values.remoteWriteTargets }}
   - alert: PrometheusRemoteWriteDown
-    expr: sum by (prometheus, url) (rate(prometheus_remote_storage_samples_failed_total[5m])) > 0 or sum by (prometheus, url) (rate(prometheus_remote_storage_bytes_total[5m])) <= 0
+    expr: |
+      (
+        (rate(prometheus_remote_storage_failed_samples_total{prometheus="{{ include "prometheus.name" . }}"}[5m]) or rate(prometheus_remote_storage_samples_failed_total{prometheus="{{ include "prometheus.name" . }}"}[5m]))
+      /
+        (
+          (rate(prometheus_remote_storage_failed_samples_total{prometheus="{{ include "prometheus.name" . }}"}[5m]) or rate(prometheus_remote_storage_samples_failed_total{prometheus="{{ include "prometheus.name" . }}"}[5m]))
+        +
+          (rate(prometheus_remote_storage_succeeded_samples_total{prometheus="{{ include "prometheus.name" . }}"}[5m]) or rate(prometheus_remote_storage_samples_total{prometheus="{{ include "prometheus.name" . }}"}[5m]))
```

Comment:
Why is `prometheus_remote_storage_failed_samples_total` and `prometheus_remote_storage_succeeded_samples_total` in here? Is this an older version? At least I can't find the metrics in our Prometheus.

Also I don't get the logic behind this:
Shouldn't it be

```failed_samples_total/samples_total``` ?

Then we would get a percentage on how many are failed. Maybe you need to elaborate, what the idea is.

Reply:
yes, the older metric can go. But otherwise, it's as it should be: `failed / total `(where `total` is the addition of `successful` and `failed`).